addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/SstiScanRule.java

```diff
@@ -0,0 +1,471 @@
+import org.zaproxy.zap.extension.ascanrulesAlpha.ssti.InputPoint;
+import org.zaproxy.zap.extension.ascanrulesAlpha.ssti.ReflectedSinkPoint;
+import org.zaproxy.zap.extension.ascanrulesAlpha.ssti.SinkPoint;
+import org.zaproxy.zap.extension.ascanrulesAlpha.ssti.SstiUtils;
+import org.zaproxy.zap.extension.ascanrulesAlpha.ssti.StoredSinkPoint;
+import org.zaproxy.zap.extension.ascanrulesAlpha.ssti.TemplateFormat;
+
+/**
+ * Active Plugin for Server Side Template Injection testing and verification.
+ *
+ * @author DiogoMRSilva (2018)
+ */
+public class SstiScanRule extends AbstractAppParamPlugin {
+
+    /** Prefix for internationalised messages used by this rule */
+    private static final String MESSAGE_PREFIX = "ascanalpha.ssti.";
+
+    static final String DELIMITER = "zj";
+
+    private static final float SIMILARITY_THRESHOLD = 0.75f;
+
+    private static String[] errorPolyglots = {"<%={{={@{#{${zj}}%>", "<th:t=\"${zj}#foreach"};
+    // Innocuous versions of the polyglots may be useful in the future if more complex error
+    // detection is required
+    // "\\{*\\<\\%\\=\\{\\{\\=\\{\\@\\{\\#\\{\\$\\{zj\\}\\}\\%\\>\\*}"
+    // the first and last tag are the commentaries from twig
+    // "\\\"\\<th\\:t\\=\\$\\{zj\\}\\#\\foreach"
+
+    private static final TemplateFormat[] TEMPLATE_FORMATS = {
+        new TemplateFormat(" ", " "),
+        new TemplateFormat("{", "}"),
+        new TemplateFormat("${", "}"),
+        new TemplateFormat("#{", "}"),
+        new TemplateFormat("{#", "}"),
+        new TemplateFormat("{@", "}"),
+        new TemplateFormat("{{", "}}"),
+        new TemplateFormat("{{=", "}}"),
+        new TemplateFormat("<%=", "%>"),
+        new TemplateFormat("#set($x=", ")${x}"),
+        new TemplateFormat("<p th:text=\"${", "}\"></p>"),
+        new TemplateFormat(
+                "{", "}", "{@math key=\"%d\" method=\"multiply\" operand=\"%d\"/}"), // Dustjs
+        new DjangoTemplateFormat(),
+        new GoTemplateFormat()
+    };
+
+    private static final String[] WAYS_TO_FIX_CODE_SYNTAX = {"\"", "'", "1", ""};
+
+    // Logger instance
+    private static final Logger LOG = LogManager.getLogger(SstiScanRule.class);
+
+    /**
+     * Get the unique identifier of this plugin
+     *
+     * @return this plugin identifier
+     */
+    @Override
+    public int getId() {
+        return 90035;
+    }
+
+    /**
+     * Get the name of this plugin
+     *
+     * @return the plugin name
+     */
+    @Override
+    public String getName() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "name");
+    }
+
+    /**
+     * Give back specific plugin dependencies (none for this)
+     *
+     * @return the list of plugins that need to be executed before
+     */
+    @Override
+    public String[] getDependency() {
+        return new String[] {};
+        // TODO activate when start to work properly
+        // return new String[] {"TestPersistentXSSSpider"};
```

Comment:
Is this still planned?

Reply:
I believe this is meant to happen after the PersistentXSS PR from Diogo that's still pending as well.

@DiogoMRSilva can you confirm?

Reply:
yes, it is for the sinking finding part of the persistentXSS. It could work without it just finding the "reflected" 
ones